test/GraphStoreProtocolTest.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <parser/SparqlParserHelpers.h>
+
+#include "./util/GTestHelpers.h"
+#include "./util/HttpRequestHelpers.h"
+#include "./util/TripleComponentTestHelpers.h"
+#include "SparqlAntlrParserTestHelpers.h"
+#include "engine/GraphStoreProtocol.h"
+
+namespace m = matchers;
+namespace t = ad_utility::testing;
+
+using Var = Variable;
+using TC = TripleComponent;
+
+TEST(GraphStoreProtocolTest, extractTargetGraph) {
+  // Equivalent to `/?default`
+  EXPECT_THAT(GraphStoreProtocol::extractTargetGraph({{"default", {""}}}),
+              DEFAULT{});
+  // Equivalent to `/?graph=foo`
+  EXPECT_THAT(GraphStoreProtocol::extractTargetGraph({{"graph", {"foo"}}}),
+              t::iri("<foo>"));
```

Comment:
The case `{"graph", {"foo", "bar"}}` can also be tested (multiple graphs throw in this domain.